chore: allow for overriding metro server port

seanstrom opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

### Summary

* This PR attempts to allow for overriding the react-native metro server port.
  * Note that we need to run `make clean` before changing the `RCT_METRO_PORT` environment variable.

seanstrom committed:
tweak: allow for overriding metro server port

status-im-auto commented:
### Jenkins Builds
<details>
<summary>Click to see older builds (31)</summary>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e | [#1](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/1/) | 2024-05-28 09:02:09 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240528-085741-ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e-pr20202-tests.log) |
| :heavy_check_mark: | ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/1/) | 2024-05-28 09:04:08 | ~6 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240528-085736-ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240528-085736-ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e-pr20202-x86_64.apk)|
| :heavy_check_mark: | ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e | [#1](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/1/) | 2024-05-28 09:04:16 | ~6 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240528-085736-ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e-pr20202-universal.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240528-085736-ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e-pr20202-universal.apk)|
| :heavy_check_mark: | ba0a8a14fc1c4e8c360f3e28d296429716ca2b6e | [#1](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/1/) | 2024-05-28 09:08:05 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/PbnjSj) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FPbnjSj)|
| | | | | | | |
| :heavy_check_mark: | 990b95d0fe8cac7380852831e3f5dbffeeff80af | [#2](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/2/) | 2024-06-05 13:54:01 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240605-134928-990b95d0fe8cac7380852831e3f5dbffeeff80af-pr20202-tests.log) |
| :heavy_check_mark: | 990b95d0fe8cac7380852831e3f5dbffeeff80af | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/2/) | 2024-06-05 13:57:21 | ~7 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240605-134925-990b95d0fe8cac7380852831e3f5dbffeeff80af-pr20202-universal.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240605-134925-990b95d0fe8cac7380852831e3f5dbffeeff80af-pr20202-universal.apk)|
| :heavy_check_mark: | 990b95d0fe8cac7380852831e3f5dbffeeff80af | [#2](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/2/) | 2024-06-05 13:57:24 | ~7 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240605-134925-990b95d0fe8cac7380852831e3f5dbffeeff80af-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240605-134925-990b95d0fe8cac7380852831e3f5dbffeeff80af-pr20202-x86_64.apk)|
| :heavy_check_mark: | 990b95d0fe8cac7380852831e3f5dbffeeff80af | [#2](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/2/) | 2024-06-05 14:02:26 | ~13 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/NSkPf3) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FNSkPf3)|
| | | | | | | |
| :heavy_check_mark: | c541e4437fe0c0208e98d46ea6e861666a3c4f0b | [#3](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/3/) | 2024-06-11 12:22:03 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240611-121738-c541e4437fe0c0208e98d46ea6e861666a3c4f0b-pr20202-tests.log) |
| :heavy_check_mark: | c541e4437fe0c0208e98d46ea6e861666a3c4f0b | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/3/) | 2024-06-11 12:23:56 | ~6 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240611-121733-c541e4437fe0c0208e98d46ea6e861666a3c4f0b-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240611-121733-c541e4437fe0c0208e98d46ea6e861666a3c4f0b-pr20202-x86_64.apk)|
| :heavy_check_mark: | c541e4437fe0c0208e98d46ea6e861666a3c4f0b | [#3](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/3/) | 2024-06-11 12:24:50 | ~7 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240611-121738-c541e4437fe0c0208e98d46ea6e861666a3c4f0b-pr20202-universal.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240611-121738-c541e4437fe0c0208e98d46ea6e861666a3c4f0b-pr20202-universal.apk)|
| :heavy_check_mark: | c541e4437fe0c0208e98d46ea6e861666a3c4f0b | [#3](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/3/) | 2024-06-11 12:27:29 | ~9 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/CtV5a5) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FCtV5a5)|
| | | | | | | |
| :heavy_check_mark: | 8925b26eb5e222b0c8ce6b86ee73d86832c66bb6 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/4/) | 2024-06-12 09:01:13 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240612-085640-8925b26eb5e222b0c8ce6b86ee73d86832c66bb6-pr20202-tests.log) |
| :heavy_check_mark: | 8925b26eb5e222b0c8ce6b86ee73d86832c66bb6 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/4/) | 2024-06-12 09:05:01 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240612-085636-8925b26eb5e222b0c8ce6b86ee73d86832c66bb6-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240612-085636-8925b26eb5e222b0c8ce6b86ee73d86832c66bb6-pr20202-x86_64.apk)|
| :heavy_check_mark: | 8925b26eb5e222b0c8ce6b86ee73d86832c66bb6 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/4/) | 2024-06-12 09:05:20 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240612-085637-8925b26eb5e222b0c8ce6b86ee73d86832c66bb6-pr20202-universal.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240612-085637-8925b26eb5e222b0c8ce6b86ee73d86832c66bb6-pr20202-universal.apk)|
| :heavy_check_mark: | 8925b26eb5e222b0c8ce6b86ee73d86832c66bb6 | [#4](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/4/) | 2024-06-12 09:06:53 | ~10 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/C5nqB8) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FC5nqB8)|
| | | | | | | |
| :heavy_check_mark: | 0e0831b6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/7/) | 2024-07-31 11:22:31 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240731-111803-0e0831-pr20202-tests.log) |
| :heavy_check_mark: | 0e0831b6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/7/) | 2024-07-31 11:24:34 | ~6 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240731-111759-0e0831-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240731-111759-0e0831-pr20202-x86_64.apk)|
| :heavy_check_mark: | 0e0831b6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/7/) | 2024-07-31 11:26:15 | ~8 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240731-111803-0e0831-pr20202-arm64-v8a.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240731-111803-0e0831-pr20202-arm64-v8a.apk)|
| :heavy_check_mark: | 0e0831b6 | [#7](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/7/) | 2024-07-31 11:30:38 | ~12 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/2oUb8R) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2F2oUb8R)|
| | | | | | | |
| :x: | 0ba8e8f9 | [#8](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/8/) | 2024-08-01 09:29:43 | ~32 sec | `android-e2e` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/8/consoleText) |
| :x: | 0ba8e8f9 | [#8](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/8/) | 2024-08-01 09:29:43 | ~32 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/8/consoleText) |
| :heavy_check_mark: | 0ba8e8f9 | [#8](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/8/) | 2024-08-01 09:33:14 | ~3 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240801-092914-0ba8e8-pr20202-tests.log) |
| :heavy_check_mark: | 0ba8e8f9 | [#8](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/8/) | 2024-08-01 09:44:39 | ~15 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/y9g9P4) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2Fy9g9P4)|
| | | | | | | |
| :x: | 7f6aee2f | [#9](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/9/) | 2024-08-01 10:15:06 | ~21 sec | `android-e2e` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/9/consoleText) |
| :x: | 7f6aee2f | [#9](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/9/) | 2024-08-01 10:15:07 | ~21 sec | `android` | [:page_facing_up:`log`](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/9/consoleText) |
| :heavy_check_mark: | 7f6aee2f | [#9](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/9/) | 2024-08-01 10:18:50 | ~3 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240801-101449-7f6aee-pr20202-tests.log) |
| | | | | | | |
| :heavy_check_mark: | 56e790f3 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/10/) | 2024-08-01 10:25:13 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240801-102025-56e790-pr20202-tests.log) |
| :heavy_check_mark: | 56e790f3 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/10/) | 2024-08-01 10:29:09 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240801-102025-56e790-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240801-102025-56e790-pr20202-x86_64.apk)|
| :heavy_check_mark: | 56e790f3 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/10/) | 2024-08-01 10:29:32 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240801-102025-56e790-pr20202-arm64-v8a.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240801-102025-56e790-pr20202-arm64-v8a.apk)|
| :heavy_check_mark: | 56e790f3 | [#10](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/10/) | 2024-08-01 10:29:52 | ~9 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/dLEB6D) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FdLEB6D)|
</details>

| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | 42c76024 | [#11](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/11/) | 2024-08-02 11:59:14 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240802-115420-42c760-pr20202-tests.log) |
| :heavy_check_mark: | 42c76024 | [#11](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/11/) | 2024-08-02 12:02:47 | ~8 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240802-115417-42c760-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240802-115417-42c760-pr20202-x86_64.apk)|
| :heavy_check_mark: | 42c76024 | [#11](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/11/) | 2024-08-02 12:03:25 | ~9 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240802-115420-42c760-pr20202-arm64-v8a.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240802-115420-42c760-pr20202-arm64-v8a.apk)|
| :heavy_check_mark: | 42c76024 | [#11](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/11/) | 2024-08-02 12:15:55 | ~21 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/NRKZcX) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FNRKZcX)|
| | | | | | | |
| :heavy_check_mark: | 427b8fba | [#12](https://ci.status.im/job/status-mobile/job/prs/job/tests/job/PR-20202/12/) | 2024-08-05 08:43:17 | ~4 min | `tests` | [:page_facing_up:`log`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240805-083851-427b8f-pr20202-tests.log) |
| :heavy_check_mark: | 427b8fba | [#12](https://ci.status.im/job/status-mobile/job/prs/job/android-e2e/job/PR-20202/12/) | 2024-08-05 08:44:59 | ~6 min | `android-e2e` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240805-083851-427b8f-pr20202-x86_64.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240805-083851-427b8f-pr20202-x86_64.apk)|
| :heavy_check_mark: | 427b8fba | [#12](https://ci.status.im/job/status-mobile/job/prs/job/android/job/PR-20202/12/) | 2024-08-05 08:46:42 | ~7 min | `android` | [:robot:`apk`](https://status-im-mobile-prs.ams3.cdn.digitaloceanspaces.com/StatusIm-Mobile-240805-083851-427b8f-pr20202-arm64-v8a.apk) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fstatus-im-mobile-prs.ams3.cdn.digitaloceanspaces.com%2FStatusIm-Mobile-240805-083851-427b8f-pr20202-arm64-v8a.apk)|
| :heavy_check_mark: | 427b8fba | [#12](https://ci.status.im/job/status-mobile/job/prs/job/ios/job/PR-20202/12/) | 2024-08-05 08:53:08 | ~14 min | `ios` | [:iphone:`ipa`](https://i.diawi.com/bTWbe4) [:calling:](https://chart.apis.google.com/chart?cht=qr&chs=400x400&chld=L%7C%0A1&chl=https%3A%2F%2Fi.diawi.com%2FbTWbe4)|

seanstrom committed:
replace 8081 with RCT_METRO_PORT

jakubgs commented on Makefile:
```diff
@@ -437,7 +437,7 @@ android-clean: ##@prepare Clean Gradle state
 android-ports: export FLOWSTORM_PORT := 7722
 android-ports: export TARGET := android-sdk
```
You need a default, and you need to use lazy assignment(`?=`) to make it overridable.
```suggestion
android-ports: export RCT_METRO_PORT ?= 8081
android-ports: export FLOWSTORM_PORT ?= 7722
android-ports: export TARGET := android-sdk
```

jakubgs commented on scripts/check-metro-shadow-process.sh:
```diff
@@ -6,7 +6,7 @@ if pgrep -f 'shadow-cljs watch mobile' > /dev/null; then
 fi
 
-if pgrep -f 'react-native start' > /dev/null; then
+if pgrep -f "react-native start --port=$RCT_METRO_PORT" > /dev/null; then
```
You need a default:
```suggestion
if pgrep -f "react-native start --port=${RCT_METRO_PORT:-8081}" > /dev/null; then
```

jakubgs commented on scripts/run-metro.sh:
```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 
-if pgrep -f 'react-native start' > /dev/null; then
+if pgrep -f "react-native start --port=$RCT_METRO_PORT" > /dev/null; then
```
No default again:
```suggestion
if pgrep -f "react-native start --port=${RCT_METRO_PORT:-8081}" > /dev/null; then
```

jakubgs commented on scripts/run-metro.sh:
```diff
@@ -1,8 +1,8 @@
 else
     echo "Info: starting a new metro terminal"
-    react-native start --reset-cache
+    react-native start --port=$RCT_METRO_PORT --reset-cache
```
```suggestion
    react-native start --port="${RCT_METRO_PORT:-8081}" --reset-cache
```

jakubgs commented on scripts/wait-for-metro-port.sh:
```diff
@@ -4,7 +4,7 @@ set -euo pipefail
 
 while [ "${TIMEOUT}" -gt 0 ]; do
-  if ! lsof -i:8081 &> /dev/null; then
+  if ! lsof -i:RCT_METRO_PORT &> /dev/null; then
```
```suggestion
  if ! lsof -i:"${RCT_METRO_PORT:-8081}" &> /dev/null; then
```

flexsurfer commented:
hey @seanstrom still relevant?

seanstrom committed:
fix: properly override RCT_METRO_PORT while providing fallback

seanstrom committed:
tweak: allow FLOWSTORM_PORT to be overridable with a fallback when configuring android ports

# 50-character subject line
#
# 72-character wrapped longer description.
#
# * Maybe try answering these questions: *
# * What did I change? *
# * Why did I change it? *
# * How did I change it? *

seanstrom commented:
@jakubgs thanks for the review 🙏 
I've made those changes that you requested. Let me know if there's anything else I should do.

seanstrom committed:
tweak: allow for overriding the metro port when building the android app

jakubgs commented on nix/mobile/android/build.nix:
```diff
@@ -17,6 +17,7 @@
   hermesEnabled ? lib.getEnvWithDefault "ORG_GRADLE_PROJECT_hermesEnabled" "true",
   buildUrl ? lib.getEnvWithDefault "ORG_GRADLE_PROJECT_buildUrl" null,
   statusGoSrcOverride ? lib.getEnvWithDefault "STATUS_GO_SRC_OVERRIDE" null,
+  reactMetroPort ? lib.getEnvWithDefault "RCT_METRO_PORT" 8081,
```
Why would you need this? Android derivation build run in a sandbox that has no network access. This is pointless.

seanstrom replied:
The intention is to forward the environment variable for `RCT_METRO_PORT` to the Gradle build step. This is because the react-native bindings will use the project property `reactNativeDevServerPort`, and I think this property needs to be defined at build time so that the react-native code knows which port to use when it has network access.

In comparison, when I didn't forward this environment variable, and I would run `make run-android`, the all the other scripts would use the correct `RCT_METRO_PORT` override, but the Android app would somehow still default to `8081`.

jakubgs replied:
Interesting, but I thought if you use Metro with `make run-metro` then we don't use the APK built by Nix.

jakubgs replied:
Oh I see, we do actually call `build-android.sh` inside of `scripts/run-android.sh`:
https://github.com/status-im/status-mobile/blob/9f412577d732a2c51e791c66955b57a42c26685a/scripts/run-android.sh#L12
And that is then started:
https://github.com/status-im/status-mobile/blob/9f412577d732a2c51e791c66955b57a42c26685a/scripts/run-android.sh#L15-L17
That's why it matters what is set for APK built by the derivation.

jakubgs reviewed (changes requested):
Please undo changes in `nix/mobile/android/build.nix`, rest looks good.

seanstrom commented on nix/mobile/android/build.nix:
```diff
@@ -158,6 +161,7 @@ in stdenv.mkDerivation rec {
       --no-build-cache \
       --parallel \
       -Dmaven.repo.local='${deps.gradle}' \
+      -PreactNativeDevServerPort=${toString reactMetroPort} \
       assemble${gradleBuildType}
```
Highlighting this part here:
* This `-PreactNativeDevServerPort` will configure the react-native build config to use the port override, otherwise it will default to `8081` and not connect to the metro server.

jakubgs replied:
If that's the case then what's the point of setting the `RCT_METRO_PORT` env variable?

seanstrom replied:
I think I was being to overly cautious with re-exporting the variable, but it doesn't seem we need it, so I've removed it.

jakubgs replied:
Cool.

jakubgs commented on nix/mobile/android/build.nix:
```diff
@@ -106,6 +108,7 @@ in stdenv.mkDerivation rec {
     export ANDROID_NDK_ROOT="${androidPkgs.ndk}"
 
     export STATUS_NIX_MAVEN_REPO="${deps.gradle}"
+    export RCT_METRO_PORT=${toString reactMetroPort}
```
I don't think you need this, the setting of `RCT_METRO_PORT` in the context of derivation arguments should be enough.

jakubgs replied:
If we are setting `-PreactNativeDevServerPort` then why do we even need the `RCT_METRO_PORT` env var for?

seanstrom replied:
Yeah good point, I tried removing these extra `RCT_METRO_PORT` variables and everything seems to be working well.

seanstrom committed:
tweak: unneeded variables for RCT_METRO_PORT in build.nix
